mantidproject/mantid
# Open Recent Scripts in workbench

**Description of work.**
Added a Recently Closed Scripts sub-menu to the File menu in workbench, it shows all scripts that have recently been closed in descending order.



**To test:**
- Open workbench
- Check File->Open Recently Closed Scripts and ensure that it shows a default no scripts found
- Then open 10 scripts and save them with a memorable name, then close them. Ensure that File-> Open Recently Closed Scripts shows all closed scripts, remember the last value in the list.
- Add another script and save with a memorable name then close it, ensure that it shows up in the list at the top, and the bottom of the list from the previous step has disappeared.
- Close Workbench
- Open Workbench
- Ensure all the scripts that were shown previously are shown again in the same order


Fixes #29032 




---

#### Reviewer ####

Please comment on the following ([full description](http://developer.mantidproject.org/ReviewingAPullRequest.html)):

##### Code Review #####

- Is the code of an acceptable quality?
- Does the code conform to the [coding standards](http://developer.mantidproject.org/Standards/)?
- Are the unit tests small and test the class in isolation?
- If there is GUI work does it follow the [GUI standards](http://developer.mantidproject.org/Standards/GUIStandards.html)?
- If there are changes in the release notes then do they describe the changes appropriately?

##### Functional Tests #####

- Do changes function as described? Add comments below that describe the tests performed?
- Do the changes handle unexpected situations, e.g. bad input?
- Has the relevant (user and developer) documentation been added/updated?

Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.
